### PR TITLE
Location and platform default fixes

### PR DESCRIPTION
**Pull request by orbl**
Makes it possible to ...
... select a platform without specifying vm-size or location
... write either` radnet init -p az` or `radnet init -p azure` and get an ephemeral network in Azure.
... get an error if one provides an invalid platform name.

Only intended to be a quick fix. Tested locally to work for both azure and gcp. 

Untested: 
... Changed default vm to n1-standard-2

### CODE REVIEW

File: client/radnet/util/network.py [hunk 1]
```diff
@@ -112,14 +115,22 @@ def write_terraform_platform_files(platform, args):
...
         sys.exit("Platforms authentication file not found for {0}".format(args['network']))
     credentials = util.load_json_from_file(get_networks_auth_file())
     args['radix_azurecr_token'] = credentials["azure_cr"]
 
     # Azure
-    if platform == PLATFORM_AZURE:
+    if platform in PLATFORM_AZURE:
+        
+        # Sets platform default if nothing specified by user.
+        if args["vm_size"] is None:
```
Comment (x-wf) [inline client/radnet/util/network.py]: Can we place this before the if elif block, so we don't have to repeat the code?

```
if args["vm_size"] is None:
        args["vm_size"] = platform == PLATFORM_AZURE ? \
            DEFAULT_VM_SIZE_AZURE : \
            DEFAULT_VM_SIZE_GCP
if args["location"] is None:
        args["location"] = platform == PLATFORM_AZURE ? \
            DEFAULT_NETWORK_LOCATION_AZURE : \
            DEFAULT_NETWORK_LOCATION_GCP
```

File: client/radnet/util/network.py [hunk 2]
```diff
@@ -147,9 +158,15 @@ def write_terraform_platform_files(platform, args):
             "path": os.path.join("terraform", "azure", "base"),
             "file": "az_nodes.j2.tf",
             "args": args})
+    elif platform == PLATFORM_GCP:
```
Comment (x-wf) [inline client/radnet/util/network.py]: So we can remove this part 👍

**Comment by blenessy** [comment]
@x-wf change proposals make sense, apart from that LGTM.

File: client/radnet/util/network.py (hunk 2 above)
Comment (orbl) [reply client/radnet/util/network.py]: Why do we want to get rid of it? The radnet cli does not support multiple platforms at once. So the three possible situations are:

```(Pseudo code)
if Azure
    do azure stuff
elif gcp 
    do gcp suff
else
    Inform the user they selected an invalid platform```

File: client/radnet/util/network.py (hunk 1 above)
Comment (orbl) [reply client/radnet/util/network.py]: Why would we want to break it from the branching logic? In the method, we branch depending on if it is azure or GCP we are handling. All other code related to either of those is within the if elif blocks. 

By breaking them out as you suggest we have three places where the logic branches based on the same criteria, instead of one.

I agree that this method is hard to follow in general and contains a lot of code duplication. But a better way to resolve that is to have a separate refactoring task for later. This pull request is too small in scope for that.

Comment (x-wf) [reply client/radnet/util/network.py]: Not sure what you mean with _break it from the branching logic_, but what I recommended was for you to place the code above the code blocks for each platform as it is not needed. 

At the moment, we only support Azure and GCP, as the code above uses a ternary operator to validate and set the correct variable depending on the platform.

In a nutshell, instead of doing **if elif**, your setting the variable based on a condition in a one liner. This won't break the usage, as you'll still continue to validate the platform.

File: client/radnet/util/network.py (hunk 2 above)
Comment (x-wf) [reply client/radnet/util/network.py]: Not get ride of the **elif** line, but the lines below 
```
if args["vm_size"] is None:
     args["vm_size"] = DEFAULT_VM_SIZE_GCP
....
```

Comment (orbl) [reply client/radnet/util/network.py]: Ah, that makes more sense.

File: client/radnet/util/network.py (hunk 1 above)
Comment (orbl) [reply client/radnet/util/network.py]: For reference only: Have had a separate talk with James about this and agreed that an acceptable way to resolve this is to write a validation method that can be called for each platform.

Comment (blenessy) [reply client/radnet/util/network.py]: cool, glad that you guys found a good solution.

Comment (orbl) [reply client/radnet/util/network.py]: I've broken out the default setting into a method. Ready to merge?

**Comment by x-wf** [comment]
LGTM
